### PR TITLE
macOS-specific changes, new artwork

### PR DESCRIPTION
**Pull request by stuartro**
Hi,

Not sure if the changes on my fork will be of interest to you, but perhaps take a look.

Changes are summarized on [my freeplane page](https://github.com/stuartro/freeplane) on the `1.8.x_sr` branch. Should merge cleanly—for whatever bits you might be interested in. 

More than likely my changes are not optimal (given my limited understanding of Freeplane’s source code / architecture), so take them with a pinch of salt. If nothing else, perhaps they can be useful as “starters” for future improvements. The artwork is likely to be a contentious issue, I know, so feel free to ignore.

Regards,
Stuart.

### CODE REVIEW

**Comment by dpolivaev** [comment]
Hey Stuart,

thank you very much for your pull request and for all ideas you have implemented. I have just have a chat with other Freeplane devs. We need some time to get through all your suggestions. I am going to come back to you after I get feedback from them.

Could you clarify if you have the copyright for the artwork and if we may use it in case we decide to change?

Best regards,
Dimitry

**Comment by stuartro** [comment]
Hi Dimitry,

 

Thanks for getting back to me. Looking forward to hearing back from you once you’ve had a chance to discuss with the other Freeplane devs.

 

RE LICENSING / COPYRIGHT

The icon is a creation of my own—both the older (pre-Big Sur) icon and the newer (Big Sur-style) icon. I’m perfectly happy to add a license statement (probably “Unlicense” or “APL v2” or something) allowing it to be freely used.


The splash screen is also my own creation, built upon this image (see attached file):



The image above was downloaded from ShutterStock (image ID: 1424938844) about 3 years ago using a free ShutterStock trial. The free trial allows up to 10 images to be downloaded and reused under the Standard License, which permits usage as follows:

ShutterStock blog post on licensing of images from free trial
“The 10 images you download during your 30-day trial are watermark-free and covered by Shutterstock’s popular Standard license. That means you can use your free images for personal and commercial projects across all kinds of mediums: books, TV, magazines, web video, advertising, film, and more.”

Ask Shutterstock: Image Usage
“You may use Shutterstock images for a variety of online and electronic projects. A Standard License lets you use Shutterstock images:
On websites, online advertising, social media, mobile advertising, mobile apps, software, e-cards, e-books, email marketing, and other online media, with no limit on viewers or distribution.
…”
 

So, in short, I have no concerns about Freeplane using the artwork—and will attest to as much if you want to use the artwork—and there are no legal limitations (that I’m aware of) on use of the image used as background in the splash screen.

 

Regards,

Stuart.

**Comment by stuartro** [comment]
Hi Dimitry,

 

Further to my last email, I have uploaded the Sketch file containing (most) of the custom SVG “node” icons to GitHub — see https://github.com/stuartro/freeplane/blob/1.8.x_sr/artwork/icons/Freeplane%20Icons.sketch. The same “usage rights” as mentioned in my email below (i.e. you can freely incorporate / use as you see fit).

 

On that front, you’ll notice that I have changed the “Local Link” and “Link” marker icons (i.e. the small green and red arrows) to be more subtle, as shown in the screenshot below in the hyperlinks (e.g. the 3 links shown under “Image used in splash screen” node). Also, the regular-expression-based hyperlink icon-decoration is shown—where hyperlinks to paths have a folder icon, hyperlinks to https://... have globe-with-lock, hyperlinks ending with ….sketch have the Sketch logo as icon, etc. 

 

 

These icons are embedded into the main source tree (as SVGs) and the Sketch files are located in https://github.com/stuartro/freeplane/tree/1.8.x_sr/artwork

—see Link.sketch and LinkLocal.sketch, etc.

 

I have other “node” icons (all SVG) in development—my goal is to have a complete set of “aesthetically pleasing” icons (all SVG) to replace the stock icons that ship with Freeplane—and will be happy to share as/when they are ready.

 

Regards,

Stuart.

 

From: "stuartro2@icloud.com" <stuartro2@icloud.com>
Date: Saturday, November 21, 2020 at 7:25 PM
To: freeplane/freeplane <reply@reply.github.com>
Subject: Re: [freeplane/freeplane] macOS-specific changes, new artwork (#176)

 

Hi Dimitry,

 

Thanks for getting back to me. Looking forward to hearing back from you once you’ve had a chance to discuss with the other Freeplane devs.

 

RE LICENSING / COPYRIGHT

The icon is a creation of my own—both the older (pre-Big Sur) icon and the newer (Big Sur-style) icon. I’m perfectly happy to add a license statement (probably “Unlicense” or “APL v2” or something) allowing it to be freely used.



The splash screen is also my own creation, built upon this image (see attached file):




The image above was downloaded from ShutterStock (image ID: 1424938844) about 3 years ago using a free ShutterStock trial. The free trial allows up to 10 images to be downloaded and reused under the Standard License, which permits usage as follows:


ShutterStock blog post on licensing of images from free trial
“The 10 images you download during your 30-day trial are watermark-free and covered by Shutterstock’s popular Standard license. That means you can use your free images for personal and commercial projects across all kinds of mediums: books, TV, magazines, web video, advertising, film, and more.”


Ask Shutterstock: Image Usage
“You may use Shutterstock images for a variety of online and electronic projects. A Standard License lets you use Shutterstock images:
On websites, online advertising, social media, mobile advertising, mobile apps, software, e-cards, e-books, email marketing, and other online media, with no limit on viewers or distribution.
…”
 

So, in short, I have no concerns about Freeplane using the artwork—and will attest to as much if you want to use the artwork—and there are no legal limitations (that I’m aware of) on use of the image used as background in the splash screen.

 

Regards,

Stuart.

**Comment by dpolivaev** [comment]
Hey,

we may not use Shutterstock image, but your work is not limited to it. So I decided to give it a try.
I use Linux for development.

Unfortunately, Freeplane throws plenty of exceptions on start. If you like we could schedule a remote session with screen sharing where we can talk about everything and decide if we want to get it work.

Would you be interested?

Just two things which I had to fix to get your version at least compile and start:

You commented out an if condition in mac.dist.gradle. Unfortunately, the code does not work on my computer, so I had to put it back.
You changed a line in freeplane.sh. Unfortunately, the new content is not correct in many relations and I had to revert it too.

With this two changes I could run Freeplane from terminal. I was shown an exception thrown many times:

```
/home/dimitry/.config/freeplane/1.8.x/linkDecoration.ini (No such file or directory)
java.lang.RuntimeException: /home/dimitry/.config/freeplane/1.8.x/linkDecoration.ini (No such file or directory)
	at org.freeplane.features.decoration.LinkDecorationConfig.loadRules(LinkDecorationConfig.java:101)
	at org.freeplane.features.decoration.LinkDecorationConfig.getRules(LinkDecorationConfig.java:50)
	at org.freeplane.features.decoration.NodeViewDecorator.addIconsForLink(NodeViewDecorator.java:57)
	at org.freeplane.features.decoration.NodeViewDecorator.getIconsForLink(NodeViewDecorator.java:36)
	at org.freeplane.view.swing.map.MainView.addIconsBasedOnLinkType(MainView.java:438)
	at org.freeplane.view.swing.map.MainView.addLinkDecorationIcons(MainView.java:418)
	at org.freeplane.view.swing.map.MainView.updateIcons(MainView.java:409)
	at org.freeplane.view.swing.map.NodeView.update(NodeView.java:1439)
	at org.freeplane.view.swing.map.NodeViewFactory.updateNewView(NodeViewFactory.java:175)
	at org.freeplane.view.swing.map.NodeViewFactory.newNodeView(NodeViewFactory.java:146)
	at org.freeplane.view.swing.map.NodeView.addChildView(NodeView.java:828)
	at org.freeplane.view.swing.map.NodeView.addChildViews(NodeView.java:823)
	at org.freeplane.view.swing.map.NodeViewFactory.updateNewView(NodeViewFactory.java:177)
	at org.freeplane.view.swing.map.NodeViewFactory.newNodeView(NodeViewFactory.java:146)
	at org.freeplane.view.swing.map.NodeView.addChildView(NodeView.java:828)
	at org.freeplane.view.swing.map.NodeView.addChildViews(NodeView.java:823)
	at org.freeplane.view.swing.map.NodeView.treeStructureChanged(NodeView.java:1391)
	at org.freeplane.view.swing.map.NodeView.setFolded(NodeView.java:941)
	at org.freeplane.view.swing.map.NodeView.setFolded(NodeView.java:933)
	at org.freeplane.view.swing.map.MapView.display(MapView.java:2352)
	at org.freeplane.view.swing.map.MapView$MapSelection.selectAsTheOnlyOneSelected(MapView.java:308)
	at org.freeplane.main.application.LastOpenedList.selectLastVisitedNode(LastOpenedList.java:171)
	at org.freeplane.main.application.LastOpenedList.afterViewCreated(LastOpenedList.java:192)
	at org.freeplane.view.swing.map.MapViewChangeObserverCompound.fireMapViewCreated(MapViewChangeObserverCompound.java:119)
	at org.freeplane.view.swing.map.MapViewChangeObserverCompound.mapViewCreated(MapViewChangeObserverCompound.java:93)
	at org.freeplane.view.swing.map.MapViewChangeObserverCompound$1.run(MapViewChangeObserverCompound.java:100)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.io.FileNotFoundException: /home/dimitry/.config/freeplane/1.8.x/linkDecoration.ini (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at org.freeplane.features.decoration.LinkDecorationConfig.loadRules(LinkDecorationConfig.java:71)
```

So I think that reviewing the changes without you would be too hard, but we could do it together if you like.
After getting it work on my computer I could discuss with other guys, which changes we want to apply.

Regards
Dimitry

**Comment by dpolivaev** [comment]
One more question. I was able to see the new splash screen, but I couldn't find any other icons you changed as I run the version from your branch.

**Comment by stuartro** [comment]
Hi Dimitry,

 

Thanks for getting back to me. I will happily work with you on a remote / screen-sharing session to talk through the changes.

 

The linkDecoration.ini error is likely due to you not having linkDecoration.ini in your Freeplane config folder. Attached is my latest linkDecoration.ini file. An older version is checked into Git – see /artwork/icons/linkDecoration.ini and /artwork/icons/README.txt.

 

Re Shutterstock… if there is interest in the icon (which is my own—from scratch—creation) then feel free to use the icon. I am also willing to look for alternative images that might look good in the splash screen. Let me know what sort of license / usage terms you guys need to adhere to and I’ll see what I can cook up.

 

Yes, I had to make some changes to the mac.dist.gradle and freeplane.sh to get things working… probably not the “right” way to fix things, but it was early in the morning and I was needing sleep… 😊

 

Let me know what works for you re remote session. I can use MS Teams / Zoom / Webex / Skype (in order of preference). If you use something else I’m sure I can make it work.

 

Regards,

Stuart.

**Comment by dpolivaev** [comment]
Hi Stuart,

thanks for your response.

I use MSTeams and zoom often. We need to find a time. What is your location
and your time zone? I am in Germany, so it is pretty late here. Would you
be available tomorrow between 5 pm and 9 pm CET? (Now it is 11:36 pm CET)

Don't spend any time on the topic before we actually meet.

Cheers,
Dimitry

**Comment by stuartro** [comment]
Hi Dimitry,

I’m in Toronto (EST)—and it’s now 5:50pm, 6 hours behind you. This 5pm–9pm CET == 11am–3pm EST, which works for me. Pick a time that is good for you and I’ll make it work. 

- Stuart

**Comment by dpolivaev** [comment]
How about 7 pm CET = 1 pn EST?

**Comment by stuartro** [comment]
Works for me.

**Comment by dpolivaev** [comment]
Hello, I have just sent you a zoom invite per e-mail. Have you got it?

**Comment by dpolivaev** [comment]
Obviously, you have not shown up.
Drop me an e-mail when you are available

**Comment by snhv** [comment]
The splash screen looks very professional. I hope it will replace the current one which looks, I'm sorry, like a children game.

**Comment by stuartro** [comment]
Hi Sam, 

Thanks for the feedback. 

I am doing some work (based on feedback from Dimitry) to replace the background image used in the splash screen with another image. The image I have currently used was obtained from ShutterStock (under a license that permits commercial and other redistribution), but Dimitry is concerned about use of commercial images. 

I’m presently looking for a decent non-commercial image. If I can’t find one, I will probably create one from scratch. The icon, however, is my own creation so no issues there.

**Comment by snhv** [comment]
Thanks Stuart for the update. I'm looking forward to your creations.

**Comment by dpolivaev** [comment]
I think now everything is integrated, so I close the pull request,
Many thanks.

**Comment by stuartro** [comment]
Thanks Dimitry. 

 

There have been a lot of things on the go on my side—my wife’s health, moving to another province, etc.—but once the dust settles I will be nice to chat / catch up / explore ideas.

 

Regards,

Stuart

 

From: Dimitry Polivaev ***@***.***>
Reply-To: freeplane/freeplane ***@***.***>
Date: Thursday, May 13, 2021 at 1:40 PM
To: freeplane/freeplane ***@***.***>
Cc: Stuart Robertson ***@***.***>, Author ***@***.***>
Subject: Re: [freeplane/freeplane] macOS-specific changes, new artwork (#176)

 

I think now everything is integrated, so I close the pull request,
Many thanks.
